test/parallel/test-runner-cli.js

```diff
@@ -28,10 +28,12 @@ const testFixtures = fixtures.path('test-runner');
   assert.strictEqual(child.signal, null);
   assert.strictEqual(child.stderr.toString(), '');
   const stdout = child.stdout.toString();
+
   assert.match(stdout, /ok 1 - this should pass/);
   assert.match(stdout, /not ok 2 - this should fail/);
-  assert.match(stdout, /ok 3 - subdir.+subdir_test\.js/);
-  assert.match(stdout, /ok 4 - this should pass/);
+  assert.match(stdout, /ok 3 - this should pass/);
```

Comment:
changing execution order might be acceptable (perhaps as semver-major), but I would not expect the number of executed tests to change. seems like a bug to me

Reply:
the number of tests did not change, as you created this pr to update it:
- #48593